tools/@aws-cdk/remodel/lib/template/packages/aws-cdk-lib/scripts/gen.ts

```diff
@@ -1,7 +1,50 @@
 import { generateAll } from '@aws-cdk/cfn2ts';
+import { ModuleDefinition } from '@aws-cdk/pkglint';
+import * as fs from 'fs-extra';
 import * as path from 'path';
 
-const srcDir = path.join(__dirname, '..', 'lib');
-generateAll(srcDir, {
-  coreImport: '../../core',
-});
+const awsCdkLibDir = path.join(__dirname, '..');
+const srcDir = path.join(awsCdkLibDir, 'lib');
+
+main()
+  .then(() => process.exit(0))
+  // eslint-ignore-next-line no-console
+  .catch(console.error)
+
+async function main() {
+  const generated = await generateAll(srcDir, {
+    coreImport: '../../core',
+  });
+
+  addExportsToPackageJson(generated);
+  addModulesToIndexFile(generated);
```

Comment:
I would collapse these into a single function to avoid iterating twice. Small optimization but easy enough.